### PR TITLE
mini update 8: describe content discovery workflow

### PR DESCRIPTION
**Pull request by pmengelbert**
Builds upon pr #167 

The relevant diff can be found here:

https://github.com/bloodorangeio/distribution-spec/compare/mini-update-7...bloodorangeio:mini-update-8?expand=1

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -79,15 +79,256 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+```
+
+The `<location>` is a pullable manifest URL.
+
+An attempt to pull a nonexistent repository MUST return response code `404 Not Found`
 
 #### Content Discovery
 
-TODO: describe the Content Discovery category and the high-level details
+Content Discovery provides a way of discovering which tags are available for a given repository.
```
Comment (jdolitsky) [inline spec.md:291]: The category should encapsulate more. Maybe make more general, and say "Currently, the only functionality provided by this workflow is the ability to discover tags ..."

File: spec.md [hunk 2]
```diff
@@ -79,15 +79,256 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+
+To fetch the list of tags, perform a `GET` request to a path in the following format:
+`/v2/<name>/tags/list`
+
+`<name>` is the namespace of the repository. Assuming a repository is found, Content Discovery endpoints MUST return a
```
Comment (jdolitsky) [inline spec.md:296]: s/Content Discovery Endpoints/this request/

File: spec.md [hunk 3]
```diff
@@ -79,15 +79,256 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+```
+
+`<name>` is the namespace of the repository, and `<tag1>`, `<tag2>`, and `<tag3>` are each tags on the repository.
+
+In addition to fetching the whole list of tags, a subset of the tags can be fetched by providing the `n` query parameter.
+In this case, the path will look like the following:
+`/v2/<name>/tags/list?n=<int>`
+
+`<name>` is the namespace of the repository, and `<int>` is an integer specifying the number of tags requested. The response
+to such a request MAY return fewer than `<int>` results, but ONLY when the total number of tags attached to the repository
```
Comment (jdolitsky) [inline spec.md:319]: ONLY does not need to be capitalized. Stop yelling at me

File: spec.md [hunk 4]
```diff
@@ -79,15 +79,256 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+is less than `<int>`. Otherwise, the response MUST include `<int>` results. Without the `last` query parameter (described
+next), the list returned will start at the beginning of the list and include `<int>` results. As above, the tags MUST be
+in lexical order.
+
+The `last` query parameter provides further means for limiting the number of tags. It is used exclusively in combination with the
+`n` parameter:
+`/v2/<name>/tags/list?n=<int>&last=<tagname>`
```
Comment (jdolitsky) [inline spec.md:326]: Use triple ticks to put on newline (like others)

File: spec.md [hunk 5]
```diff
@@ -79,15 +79,256 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+
+`<name>` is the namespace of the repository, `<int>` is the number of tags requested, and `<tagname>` is the *value* of
+the last tag. `<tagname>` MUST NOT be an index, but rather it MUST be a proper tag. A request of this sort will return
```
Comment (jdolitsky) [inline spec.md:329]: Maybe "numerical index" would make this more clear

File: spec.md [hunk 6]
```diff
@@ -79,15 +79,255 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+`<name>` is the namespace of the repository, and `<tag1>`, `<tag2>`, and `<tag3>` are each tags on the repository.
+
+In addition to fetching the whole list of tags, a subset of the tags can be fetched by providing the `n` query parameter.
+In this case, the path will look like the following:
+`/v2/<name>/tags/list?n=<int>`
+
+`<name>` is the namespace of the repository, and `<int>` is an integer specifying the number of tags requested. The response
+to such a request MAY return fewer than `<int>` results, but only when the total number of tags attached to the repository
+is less than `<int>`. Otherwise, the response MUST include `<int>` results. Without the `last` query parameter (described
+next), the list returned will start at the beginning of the list and include `<int>` results. As above, the tags MUST be
```
Comment (jzelinskie) [inline spec.md:318]: I think this sentence should just be moved into the paragraph that's dedicated to `last`. It's confusing being here, but I understand why you did it.

File: spec.md [hunk 7]
```diff
@@ -79,15 +79,255 @@ Registry providers can self-certify by submitting conformance results to [openco
...
+The `last` query parameter provides further means for limiting the number of tags. It is used exclusively in combination with the
+`n` parameter:
+```
+/v2/<name>/tags/list?n=<int>&last=<tagname>
+```
+
+`<name>` is the namespace of the repository, `<int>` is the number of tags requested, and `<tagname>` is the *value* of
+the last tag. `<tagname>` MUST NOT be a numerical index, but rather it MUST be a proper tag. A request of this sort will return
+up to `<int>` tags, beginning non-inclusively with `<tagname>`. That is to say, `<tagname>` will not be included in the
+results, but up to `<int>` tags *after* `<tagname>` will be returned. The tags MUST be in lexical order.
```
Comment (jzelinskie) [inline spec.md:330]: Content like this reads better in a table, imo.

**Comment by jdolitsky** [comment]
Closing this in favor of https://github.com/opencontainers/distribution-spec/pull/178

**Pull request closed without merging**